azurerm/internal/services/containers/kubernetes_cluster_data_source.go

```diff
@@ -1110,7 +1120,20 @@ func flattenKubernetesClusterDataSourceManagedClusterIdentity(input *containerse
 		identity["tenant_id"] = *input.TenantID
 	}
 
+	identity["user_assigned_identity_id"] = ""
+	if input.UserAssignedIdentities != nil {
+		keys := []string{}
+		for key := range input.UserAssignedIdentities {
+			keys = append(keys, key)
+		}
+		parsedId, err := msiparse.UserAssignedIdentityID(keys[0])
```

Comment:
(as below) there's a panic here if `keys` is a non-nil empty list, could we make this:

```suggestion
		if len(keys) > 0 {
			parsedId, err := msiparse.UserAssignedIdentityID(keys[0])
			if err != nil {
	 			return nil, err
	 		}
 			identity["user_assigned_identity_id"] = parsedId.ID()
		}
```